indy_node/server/action_req_handler.py

```diff
@@ -68,16 +71,25 @@ def validate(self, req: Request):
                     Roles.nameFromValue(origin_role)))
 
     def apply(self, req: Request, cons_time: int = None):
-        if req.txn_type != POOL_RESTART:
-            raise InvalidClientRequest("{} is not type of action transaction"
-                                       .format(req.txn_type))
         result = {}
         try:
-            self.restarter.handleActionTxn(req)
-            result = self._generate_action_result(req)
+            if req.txn_type == POOL_RESTART:
+                self.restarter.handleActionTxn(req)
+                result = self._generate_action_result(req)
+            elif req.txn_type == VALIDATOR_INFO:
+                result = self._generate_action_result(req)
+                result[DATA] = self.info_tool.info
+            else:
+                raise InvalidClientRequest(
+                    "{} is not type of action transaction"
+                    .format(req.txn_type))
         except Exception as ex:
-            result = self._generate_action_result(req, False, ex.args[0])
-            logger.warning("Restart is failed")
+            if isinstance(ex, InvalidClientRequest):
+                raise ex
+            result = self._generate_action_result(req,
+                                                  False,
+                                                  ex.args[0])
+            logger.warning("Operation is failed")
```

Comment:
I think it is better to move to the begging of the except block and extend message with the input req
In case of InvalidClientReq we will not have any logs